uclient-fetch.c

```diff
@@ -517,6 +535,22 @@ static int no_ssl(const char *progname)
 	return 1;
 }
 
+static bool is_valid_header(char *str)
```

Comment:
This is a good thing to check a header name but not necessary.
To keep the code simpler and smaller we can drop the check. Unless a server is ok with a header name that's not our business.
Maybe the full GNU wget or curl do the check to avoid users mistake. But here we expect that a script was tested and a user is not dumb.

Reply:
I mostly introduced this check to make sure `uclient-fetch` exists with an error and actively refuses to violate the protocol.